### ISSUE TITLE
[Phase 1] Hugging Face Transformers 라이브러리 실습

### ISSUE THREAD

**Issue by YOONPYOGitHub** [YOONPYOGitHub/knowledge-distillation#36]
## 목표
프로젝트 전반에서 사용하는 Hugging Face 생태계를 실습하며 익히기

## 기간
Phase 1 (2~3개월차) | 예상 2주

### Transformers 라이브러리 기본
- [ ] `AutoModel`, `AutoTokenizer`, `AutoConfig` 패턴 이해
- [ ] `from_pretrained()` 메커니즘 — 캐시, 다운로드 경로
- [ ] Model forward pass: input_ids → logits 흐름
- [ ] `model.eval()` vs `model.train()` 모드 차이
- [ ] `torch.no_grad()` 사용 이유와 타이밍

### Datasets 라이브러리
- [ ] `load_dataset("wikitext", "wikitext-2-raw-v1")` 실습
- [ ] Dataset 구조 파악 (train / validation / test split)
- [ ] `map()` 함수로 토크나이징 적용
- [ ] DataLoader와의 연동

### 텍스트 생성 실습
- [ ] `model.generate()` 사용법
- [ ] sampling 전략: greedy, top-k, top-p, temperature
- [ ] 동일 프롬프트로 gpt2 vs gpt2-large 출력 비교

### GPU 활용 기초
- [ ] `.to("cuda")` 로 모델/데이터 GPU 이동
- [ ] `torch.cuda.memory_allocated()` 메모리 확인
- [ ] FP16 (`torch.float16`) 로드 실습
- [ ] GPU OOM 발생 시 대처법 기초

## 실습 과제
```python
# 이런 코드를 직접 작성해보기
from transformers import AutoModelForCausalLM, AutoTokenizer
model = AutoModelForCausalLM.from_pretrained("gpt2")
tokenizer = AutoTokenizer.from_pretrained("gpt2")
inputs = tokenizer("Knowledge distillation is", return_tensors="pt")
outputs = model(**inputs)
print(outputs.logits.shape)  # [1, seq_len, 50257]
```

## 참고 자료
- `docs/model-installation-guide.md`
- `examples/` 디렉토리 코드 참고
- Hugging Face 공식 튜토리얼

## 완료 기준
- [ ] GPT-2 모델을 로드하고 logit 출력까지 독립적으로 가능
- [ ] WikiText-2를 로드하고 토크나이징하여 배치 생성 가능
- [ ] GPU에서 모델 추론 실행 가능
